docs/design/fabric-v2+/trusted_ledger.md

```diff
@@ -0,0 +1,171 @@
+# Trusted Ledger (TLCC)
+
+The purpose of TLCC is to establish a trusted view of the ledger (inside an enclave)
+and enable ECC to verify ledger state receiveed from the (untrusted) peer.
+
+
+## Requirements
+
+- ECC must verify all untrusted ledger state inputs from the peer with the help of TLCC
+    - Secure (authenticated) channel between ECC and TLCC to query metadata;
+    - TLCC must provide integrity metadata for the ledger state
+    - TLCC must provide channel metadata (chaincode definition)
+
+- TLCC must maintain a "trusted" view of the ledger
+    - TLCC must be syncronized with peer's ledger
+    - TLCC must detect transaction read/write conflicts
+    - TLCC must be able to perform identity validation (orderer, msp, etc...)
+    - TLCC must support concurrent execution of transaction processing and ledger updates
+        (ECC tx must perform read/write on a stable view of the ledger state)
+    
+
+## Fabric high-level validation 
+
+TODO pseudo code here
+
+Config validation: [source](https://github.com/hyperledger/fabric/blob/f27912f2f419c3b35d2c1df120f19585815eceb0/common/configtx/validator.go#L163)
+
+- check config sequence number increased by 1
+- check is authorized Update [source](https://github.com/hyperledger/fabric/blob/f27912f2f419c3b35d2c1df120f19585815eceb0/common/configtx/update.go#L115)
+    - verify ReadSet [source](https://github.com/hyperledger/fabric/blob/f27912f2f419c3b35d2c1df120f19585815eceb0/common/configtx/update.go#L18)
+    - verify DeltaSet [source](https://github.com/hyperledger/fabric/blob/f27912f2f419c3b35d2c1df120f19585815eceb0/common/configtx/update.go#L68)
+        - for each item validate policy [source](https://github.com/hyperledger/fabric/blob/f27912f2f419c3b35d2c1df120f19585815eceb0/common/policies/policy.go#L133)
+
+Lifecycle validation:
+
+
+Default validation:
+
+- Validating identities that signed the transaction
+    - read/write check?
+- Verifying the signatures of the endorsers on the transaction
+    - can endorse?
+- Ensuring the transaction satisfies the endorsement policies of the namespaces of the corresponding chaincodes.
+
+
+## Approach
+
+Restrict Fabric functionality:
+
+- keep Fabric validation changes in sync with TLCC
+- Reduces validation logic to a minimum
+    - minimizes catch-up with Fabric updates
+- Restrict the notion of endorsement policy; 
+- Simplify chaincode versioning (TODO)
+    - No chaincode updates for now
+- Support for default MSP only
+- Maintaining FPC chaincode state only is sufficient
+- Validate "normal" Enclave Registry (ERCC) transactions; however, ERCC comes with "hard-coded" endorsement policy
+
+
+Any transactions/blocks with unsupported features are ignored/aborted
```

Comment:
somewhere we definitely have to be more explicit where we abort and, in particular, where we (safely can) ignore. My feeling is that by default we should abort and ignore only for explicit cases where we know it's safe to ignore ..